main.go

```diff
@@ -164,13 +178,28 @@ func runServer(url string) {
 	log.Printf("Available tool: APR Calculator")
 	log.Printf("Health check available at /health")
 
-	// Start the HTTP server with logging handler
-	if err := http.ListenAndServe(url, handlerWithLogging); err != nil {
-		log.Fatalf("Server failed: %v", err)
+	go func() {
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatalf("Server failed: %v", err)
+		}
+	}()
```

Comment:
Implementing graceful shutdown is crucial for production services. It ensures that in-flight requests are completed before the server stops, preventing dropped connections during deployments or restarts.